enumerables.rb

```diff
@@ -0,0 +1,201 @@
+# rubocop:disable
+module Enumerable
+  # my_each_Enumberable method
+  def my_each
+    return to_enum unless block_given?
+
+    arr = to_a
+    i = 0
+    while i < arr.length
+      yield(arr[i])
+      i += 1
+    end
+    self
+  end
+
+  # my_each_with_index Enumberable method
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    index = 0
+    my_each do |item|
+      yield(item, index)
+      index += 1
+    end
+
+    self
+  end
+
+  # my_select Enumberable method
+  def my_select
+    return to_enum unless block_given?
+
+    array = []
+
+    my_each { |num| array << num if yield(num) }
+
+    array
+  end
+
+  # my_all Enumberable method
+  def my_all?(argument = nil)
+    if argument
+      my_each { |element| return false unless argument === element } # rubocop:disable Style/CaseEquality
+    elsif block_given?
+      my_each { |element| return false unless yield(element) }
+    else
+      my_each { |element| return false unless element }
+    end
+    true
+  end
+
+  # my_any Enumberable method
+  def my_any?(argue = nil)
```

Comment:
Your `my_any` should return false if at least one of the collection is not true when no block or argument is given
Currently, your method does not handle this scenario accurately